mongo/client_encryption.go

```diff
@@ -41,14 +42,18 @@ func NewClientEncryption(keyVaultClient *Client, opts ...*options.ClientEncrypti
 	db, coll := splitNamespace(ceo.KeyVaultNamespace)
 	ce.keyVaultColl = ce.keyVaultClient.Database(db).Collection(coll, keyVaultCollOpts)
 
+	kmsProviders, err := transformBsoncoreDocument(bson.DefaultRegistry, ceo.KmsProviders)
```

Comment:
Right now, this is just using `bson.DefaultRegistry`. I can add a `Registry` option to `ClientEncryptionOptions` to make it consistent with the `Client` type if anyone thinks that would be useful. We can also file a separate ticket to add that or just wait until we see a need for it.

Reply:
I considered that, but I'm not sure if it makes sense to use that Client's registry. Generally, registries should be set for each top-level thing and then inherited from there on. For example, you can set a registry on Client and it will be inherited by Database/Collection. ClientEncryption isn't really inherited from Client, though, so I wasn't sure if that would be a weird abstraction

Filed GODRIVER-1781 to consider this further so we don't block this PR on it.